Lit action updates (#327)
LIT-Protocol/docs: docs/sdk/serverless-signing/combining-signatures.md

import FeedbackComponent from "@site/src/pages/feedback.md";

# Signing within a Lit Action

:::info
    Only available on the Lit networks:
        - `datil`
        - `datil-test`
        - `datil-dev` 
:::

## Overview

> When you [sign a message with Lit](../serverless-signing/quick-start#sign-a-transaction.md), signature shares are typically combined client-side. However, the `signAndCombineEcdsa` function allows you to combine signature shares directly within a Lit Action, which is useful for performing operations where heightened security or privacy are paramount. The signature shares will remain within the confines of each Lit node's [Trusted Execution Environment (TEE)](../../resources/how-it-works#1-lit-nodes.md) without ever being exposed to the outside world. 

Reviewer: ```suggestion
When you [sign a message with Lit](../serverless-signing/quick-start#sign-a-transaction.md), signature shares are typically combined client-side. However, the `signAndCombineEcdsa` function allows you to combine signature shares directly within a Lit Action, which is useful for when you want to make use of the signed data with your Lit Action e.g. submitting a signed transaction. The signature shares will remain within the confines of each Lit node's [Trusted Execution Environment (TEE)](../../resources/how-it-works#1-lit-nodes.md) without ever being exposed to the outside world. 
```

> When you call the `signAndCombineEcdsa` function, signature shares are collected from each Lit node before being combined on a *single* node. The following doc will show you how you can use this functionality for arbitrary message signing, as well as how to sign a blockchain transaction using ethersJS. A complete example project is linked at the bottom of this page. 

Reviewer: ```suggestion
When you call the `signAndCombineEcdsa` function, signature shares are collected from each Lit node before being combined on a *single* node. The following code example will show how you can use this functionality for arbitrary message signing, as well as how to sign a blockchain transaction using ethers.js. A complete code example is linked at the bottom of this page. 
```

## Signing a message

The following Lit Action uses `signAndCombineEcdsa` to sign the message 'hello world'. 

```js

> const code = (async () => {

Reviewer: ```suggestion
const code = `(async () => {
```

    sigName,
  });
  
  // Set the response from the action as the signature share which will not need to be combined on the client
  Lit.Actions.setResponse({ response: JSON.stringify(signature) });

> })();

Reviewer: ```suggestion
})();`
```

    }
});

console.log("response from signing in a transaction: ", res);
```

## Signing a Transaction

> The following Lit Action uses `EthersJS` to serialize and sign a transaction (combining signature shares from the Lit nodes) before sending it back to the client where it can be broadcasted to chain. 

Reviewer: ```suggestion
The following Lit Action uses `ethers.js` to serialize and sign a transaction (combining signature shares from the Lit nodes) before sending it back to the client where it can be broadcasted to chain. 
```

```js

> const code = (async () => {

Reviewer: ```suggestion
const code = `(async () => {
```

    response: signature
  });
})();

> ;

Reviewer: ```suggestion
`;
```

console.log("result from singing in a lit action", res);
```

## Complete Example: Signing Blockchain Transactions

> You can find a full example using the `signAndCombineEcdsa` function to sign and send a blockchain transaction on the Sepolia testnet [here](https://github.com/LIT-Protocol/developer-guides-code/tree/wyatt/sign-and-combine/sign-and-combine-ecdsa).

Reviewer: I think @awisniew207 is changing this to submit a tx on Yellowstone instead of Sepolia
